src/bootloader/bootloaderlite.cc

```diff
@@ -93,17 +95,8 @@ BootloaderLite::VersionStrRes BootloaderLite::getDeploymentVersion(const std::st
 std::string BootloaderLite::getVersion(const std::string& deployment_dir, const std::string& hash,
                                        const std::string& ver_file) {
   try {
-    std::string file;
-    for (auto& p : boost::filesystem::directory_iterator(deployment_dir)) {
-      std::string dir = p.path().string();
-      if (!boost::filesystem::is_directory(dir)) {
-        continue;
-      }
-      if (boost::algorithm::contains(dir, hash)) {
-        file = dir + ver_file;
-        break;
-      }
-    }
+    std::string file = getTargetDir(deployment_dir, hash) + ver_file;
```

Comment:
`file` won't be empty even if `getTargetDir` returns empty string, so the `if` at line 100 won't work.
I suppose we need to check return value of the `getTargetDir()` call at first, and then append `ver_file`.